### ISSUE TITLE
import/no-unresolved errors after updating to Atom 1.21

### ISSUE THREAD

**Issue by edmundito** [AtomLinter/linter-eslint#1034]
## Issue Type
Bug

## Issue Description
After I updated atom, linter-eslint stopped working correctly. It seems to be ignoring the .eslintrc.js and .eslintignore file in my project, and it's unable to resolve modules, therefore it is displaying incorrect linter errors.

Our project is not using the latest eslint, but it's been working fine until now.

Debug Info:
```
Atom version: 1.21.0
linter-eslint version: 8.3.2
ESLint version: 3.8.1
Hours since last Atom restart: 0.1
Platform: darwin
Using local project ESLint from: /Users/edmundito/project/node_modules/eslint
Current file's scopes: [
  "source.js.jsx",
  "punctuation.terminator.statement.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by Arcanemagus** [comment]
Try making sure you have _completely_ exited Atom and starting it back up again, it's working just fine here...

**Comment by edmundito** [comment]
I've tried a number of things including: Restarting, clearing the atom cache, ensuring the atom helpers were not running in the background, uninstalling and reinstalling linter-eslint, running eslint on the CLI, but no luck yet...

**Comment by edmundito** [comment]
I investigated a bit further and it seems specific to `import/no-unresolved` rule showing an error even for the simplest import (exmaple, `import _ from 'lodash'`), the rest of the rules seem to work fine. The project is using the following:

babel-plugin-module-resolver
eslint-import-resolver-babel-module
eslint-plugin-import

**Comment by Arcanemagus** [comment]
So far in _every_ case that I've investigated this has been due to a bug in the `eslint-import-resolver-*` module. If you upload a .zip of your project or create a test repo I can verify if that's the case again here.

**Comment by Arcanemagus** [comment]
Generally this is due to an error being thrown from the resolver configuration that is hidden by `eslint-plugin-import`.

**Comment by edmundito** [comment]
Thanks for pointing me in the right direction.

I had a chance to look at this further and there seems some kind of case-insensitive issue between Atom's filesystem module and our version of `eslint-plugin-import`. I found a workaround in the meantime.

**Comment by IanVS** [comment]
@edmundito do you still believe this is an issue with `linter-eslint`, or can the issue be closed?

**Comment by edmundito** [comment]
You can close the issue, this turned out to not be related to linter-eslint.

**Comment by edmundito** [comment]
OK, I couldn't give up just yet... 😁 

I figured out what the problem was, it's the `TextEditor::getPath()` issue that @Arcanemagus reported in: https://github.com/atom/atom/issues/14402

After I upgraded Atom, I must have somehow opened the project in the wrong case. This affects the eslint-plugin-import's case insensitivity check. By putting a breakpoint in the linter-eslint code before sending the filePath to the eslint cli worker and correcting the path, i noticed that it would pass correctly, then I noticed the path in the wrong case in the Atom titlebar. The solution was to close the Atom window and re-open the project with the correct path.

In newer versions of `eslint-plugin-import` you can set the rule to ignore the check:

```js
'import/no-unresolved': ['error', { caseSensitive: false }]
```
https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unresolved.md
